include/gunrock/cuda/context.hxx

```diff
@@ -151,6 +151,9 @@ class multi_context_t {
         cudaDeviceEnablePeerAccess(ctx_peer->ordinal(), 0);
       }
     }
+    
+    auto ctx0 = get_context(0);
+    cudaSetDevice(ctx0->ordinal());
```

Comment:
https://github.com/bkj/essentials/blob/6bdcb4d50b0bd3f534d377d340dfadd10593a464/include/gunrock/cuda/context.hxx#L66

So, the non-blocking stuff we were talking about is changing the line above to a non-blocking stream creation;
```cpp
cudaStreamCreateWithFlags(&_stream, cudaStreamNonBlocking);
```

https://github.com/bkj/essentials/blob/6bdcb4d50b0bd3f534d377d340dfadd10593a464/include/gunrock/cuda/context.hxx#L68
Similarly, events always have timers in them, we can disable the timers to improve performance;
```cpp
cudaEventCreateWithFlags (&_event, cudaEventDisableTiming);
```